### ISSUE TITLE
Collection creation returns a 200 on a deleted collection instead of a 201

### ISSUE THREAD

**Issue by Natim** [mozilla-services/cliquet#367]
When running the loadtest, this always occurs 3 times in a raw with the memory backend.

    echo '{"data": {}}' | http PUT http://localhost:8888/v1/buckets/bucket-9c323df5-856a-427c-993c-3a38c0260289\
        --auth 'd9ad49c48db646b5a4c2511cfe855bc7:secret'
    echo '{"data": {}, "permissions": {"record:create": ["system.Authenticated"]}}' | \
        http PUT http://localhost:8888/v1/buckets/bucket-9c323df5-856a-427c-993c-3a38c0260289/collections/tasks \
        --auth 'd9ad49c48db646b5a4c2511cfe855bc7:secret'

returns a 200 instead of a 201

Same results with:

  - ``d0eff5fedff946e6802cbbd6feb22b68:secret`` and ``/buckets/bucket-77419427-128f-4d8f-b02a-8a2105fddcac/collections/tasks``
  - ``972fc9c63b214b8c90093c1e7efc5ca0:secret`` and ``/buckets/bucket-2d9388fa-82a6-4774-b0f8-6ca67b8583db/collections/tasks``

**Comment by Natim** [comment]
- [ ] Update the kinto loadtests - https://github.com/mozilla-services/kinto/blob/118-test-tutorial/loadtests/loadtest/tutorial.py#L182

**Comment by Natim** [comment]
I am using this to try to debug it:

```
diff --git a/kinto/views/buckets.py b/kinto/views/buckets.py
index b0a7371..6dd6921 100644
--- a/kinto/views/buckets.py
+++ b/kinto/views/buckets.py
@@ -37,6 +37,8 @@ def create_collection(request, bucket_id):
         collection_id = subpath.split('/')[2]
         collection_put = (request.method.lower() == 'put' and
                           request.path.endswith(collection_id))
+        if request.method.lower() == "put" and collection_id in request.path and not collection_put:
+            import pdb; pdb.set_trace()
         if not collection_put:
             subrequest = build_request(request, {
                 'method': 'PUT',
diff --git a/loadtests/loadtest/tutorial.py b/loadtests/loadtest/tutorial.py
index bd72c5e..d35cdae 100644
--- a/loadtests/loadtest/tutorial.py
+++ b/loadtests/loadtest/tutorial.py
@@ -179,10 +179,9 @@ class TutorialLoadTestMixin(object):
             auth=self.auth,
             headers={'Content-Type': 'application/json'})
         self.incr_counter("status-%s" % resp.status_code)
-        # We should always have a 201 here. See mozilla-services/cliquet#367
-        # if resp.status_code == 200:
-        #     import pdb; pdb.set_trace()
-        # self.assertEqual(resp.status_code, 201)
+        if resp.status_code == 200:
+            import pdb; pdb.set_trace()
+        self.assertEqual(resp.status_code, 201)
         record = resp.json()
         self.assertIn('record:create', record['permissions'])
         self.assertIn('system.Authenticated',
```

**Comment by leplatrem** [comment]
What makes you think that this a cliquet bug ?

**Comment by Natim** [comment]
Good point, could be a kinto bug.

**Comment by Natim** [comment]
If you want to reproduce:

```python
import requests
import uuid

collection_id = str(uuid.uuid4())
count = 0
error = 0

while True:
    resp = requests.put(
        'http://localhost:8888/v1/buckets/default/collections/%s' % collection_id,
        json={'data': {}},
        auth=('user', 'pass'),
        headers={'Content-Type': 'application/json'})
    if resp.status_code != 200:
        error += 1
    count += 1

    if count == 100:
        break

print "%s / %s errors" % (error, count)
```

**Comment by Natim** [comment]
Let me try to reproduce with asyncio first :)

**Comment by Natim** [comment]
BTW you can reproduce by decommenting: https://github.com/mozilla-services/kinto/blob/master/loadtests/loadtest/tutorial.py#L148 or/and https://github.com/mozilla-services/kinto/blob/master/loadtests/loadtest/tutorial.py#L161

**Comment by Natim** [comment]
It should be fixed with the work done about tombstones.

**Comment by leplatrem** [comment]
If not already fixed, it would look like a Kinto bug to me :)

**Comment by Natim** [comment]
Yes I tried to remove the comment in the loadtest without success. I hope we can tackle this soon enough.

**Comment by Natim** [comment]
This bug is probably related to Kinto/kinto#127 because we are recreating a deleted collection and we get a 200 instead of a 201 in that case.

**Comment by Natim** [comment]
```http
$ http PUT https://kinto.dev.mozaws.net/v1/buckets/default/collections/reproduce-error   --auth natim:
HTTP/1.1 201 Created
Access-Control-Expose-Headers: Backoff, Retry-After, Alert, Content-Length
Connection: keep-alive
Content-Length: 178
Content-Type: application/json; charset=UTF-8
Date: Wed, 16 Sep 2015 10:07:58 GMT
ETag: "1442398078684"
Last-Modified: Wed, 16 Sep 2015 10:07:58 GMT
Server: nginx/1.4.6 (Ubuntu)

{
    "data": {
        "id": "reproduce-error", 
        "last_modified": 1442398078684, 
        "schema": {}
    }, 
    "permissions": {
        "write": [
            "basicauth:df93ca0ecaeaa3126595f6785b39c408be2539173c991a7b2e3181a9826a69bc"
        ]
    }
}

$ http DELETE https://kinto.dev.mozaws.net/v1/buckets/default/collections/reproduce-error   --auth natim:
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Backoff, Retry-After, Alert, Content-Length
Connection: keep-alive
Content-Length: 78
Content-Type: application/json; charset=UTF-8
Date: Wed, 16 Sep 2015 10:08:05 GMT
Server: nginx/1.4.6 (Ubuntu)

{
    "data": {
        "deleted": true, 
        "id": "reproduce-error", 
        "last_modified": 1442398085144
    }
}

$ http PUT https://kinto.dev.mozaws.net/v1/buckets/default/collections/reproduce-error   --auth natim:
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Backoff, Retry-After, Alert, Content-Length
Connection: keep-alive
Content-Length: 178
Content-Type: application/json; charset=UTF-8
Date: Wed, 16 Sep 2015 10:08:08 GMT
ETag: "1442398088018"
Last-Modified: Wed, 16 Sep 2015 10:08:08 GMT
Server: nginx/1.4.6 (Ubuntu)

{
    "data": {
        "id": "reproduce-error", 
        "last_modified": 1442398088018, 
        "schema": {}
    }, 
    "permissions": {
        "write": [
            "basicauth:df93ca0ecaeaa3126595f6785b39c408be2539173c991a7b2e3181a9826a69bc"
        ]
    }
}
```

**Comment by leplatrem** [comment]
Oh, this becomes interesting!

This is betweeen this:
https://github.com/mozilla-services/cliquet/blob/master/cliquet/resource.py#L516-L530

And this:
https://github.com/Kinto/kinto/blob/master/kinto/views/collections.py#L57-L62

\o/

**Comment by Natim** [comment]
We probably want to remove this code: ``https://github.com/mozilla-services/cliquet/blob/master/cliquet/resource.py#L520-L525``

**Comment by leplatrem** [comment]
I would rather make sure the purge of collection deletion works as expected :)
This code was meant to fix https://github.com/mozilla-services/cliquet/issues/249 (https://github.com/mozilla-services/cliquet/commit/e5b7d83395cfb0d59dbc8b0e38885645e5af9a11)

**Comment by Natim** [comment]
Yes but don't forget about https://github.com/mozilla-services/cliquet/issues/365 discussion either.

**Comment by Natim** [comment]
In any case I think we should have a 201 from a tombstone to a record.

**Comment by leplatrem** [comment]
Fixed in Cliquet 2.10
